lib/bson/ext_json.rb

```diff
@@ -363,7 +363,7 @@ module ExtJSON
 
     module_function def map_hash(hash, **options)
       ::Hash[hash.map do |key, value|
-        if key.include? NULL_BYTE
+        if key.to_s.include? NULL_BYTE
```

Comment:
If I am reading the diff correctly, key can be of any type. If so the behavior should be:

- for strings, use `include?` directly as before
- for symbols, use `to_s.include?` as proposed now
- for everything else, do nothing
When the key is of an unhandled type I don't think it should be validated here. If there is a concern that embedded nulls would show up elsewhere, I suggest adding a validation at a point where we actually deserialize or serialize the key to a string.

Reply:
done